Zotero/Controllers/EmojiExtractor.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  EmojiExtractor.swift
+//  Zotero
+//
+//  Created by Michal Rentka on 22.11.2023.
+//  Copyright © 2023 Corporation for Digital Scholarship. All rights reserved.
+//
+
+import Foundation
+
+struct EmojiExtractor {
+    static func extractFirstContiguousGroup(from text: String) -> String? {
+        var startIndex: Int?
+        var endIndex: Int?
+
+        for (idx, character) in text.enumerated() {
+            let isEmoji = isEmoji(character: character)
+            if startIndex == nil && isEmoji {
+                startIndex = idx
+            }
+            if startIndex != nil && !isEmoji {
+                endIndex = idx
+                break
+            }
+        }
+
+        if startIndex != nil && endIndex == nil {
+            endIndex = text.count
+        }
+
+        guard let startIndex, let endIndex else { return nil }
+        return String(text[text.index(text.startIndex, offsetBy: startIndex)..<text.index(text.startIndex, offsetBy: endIndex)])
+    }
+
+    private static func isEmoji(character: Character) -> Bool {
+        guard let firstScalar = character.unicodeScalars.first else { return false }
+
+        if character.unicodeScalars.count > 1 {
+            return firstScalar.properties.isEmoji
+        }
+
+        return firstScalar.properties.isEmoji && firstScalar.value > 0x238C
```

Comment:
Is the lower limit arbitrary?

Reply:
It should be the first emoji UTF16 value. I can't use only the `isEmoji`, because it also detects some other symbols (#, *, 6) as emojis, so this check avoids incorrect detections. (I've got this from this discussion: https://stackoverflow.com/questions/30757193/find-out-if-character-in-string-is-emoji)

Reply:
You're right, that lower bound seems better, updated 41a8c98636ebd233ef491a27db3181cee453107b